### PR TITLE
Patches for generalization

### PR DESCRIPTION
**Pull request by tatyano**

### CODE REVIEW

File: libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/impl/UserAuthority.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
     private static final Logger LOG = LoggerFactory.getLogger(UserAuthority.class);
     static final String ATHENZ_PROP_PAM_SERVICE_NAME = "athenz.auth.user.pam_service_name";
 
+    final String userDomain = System.getProperty("athenz.user_domain", "user");
```
Comment (havetisyan) [inline libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/impl/UserAuthority.java]: Not sure what is the reason for this? Is there is a reason why you want to configure the domain name? Most likely you need to define your own authority as oppose to changing the default user authority.

File: servers/zms/schema/zms_server.sql [hunk 2]
```diff
@@ -37,7 +37,8 @@ CREATE TABLE IF NOT EXISTS `zms_server`.`domain` (
   INDEX `idx_modified` (`modified` ASC),
   INDEX `idx_account` (`account` ASC),
   INDEX `idx_ypmid` (`ypm_id` ASC))
-ENGINE = InnoDB;
+ENGINE=InnoDB DEFAULT CHARSET=utf8 ROW_FORMAT=DYNAMIC;
+-- Prevent ERROR 1709 (HY000)
```
Comment (havetisyan) [inline servers/zms/schema/zms_server.sql]: Can you please provide additional details as far as what is causing this error. 

btw, this is auto generated file so it cannot be modified. if necessary changes should be done in the workbench file and generate a new version of the sql script.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 3]
```diff
@@ -740,10 +740,6 @@ void initObjectStore() {
             adminUsers.add(adminUser);
         }
         
-        if (!ZMSConsts.USER_DOMAIN.equals(userDomain)) {
```
Comment (havetisyan) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java]: I think the bug here is not to remove this block. Instead we want to create both domains - userDomain and USER_DOMAIN.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSResources.java [hunk 4]
```diff
@@ -122,7 +122,8 @@ public Domain postSubDomain(@PathParam("parent") String parent, @HeaderParam("Y-
     public Domain postUserDomain(@PathParam("name") String name, @HeaderParam("Y-Audit-Ref") String auditRef, UserDomain detail) {
         try {
             ResourceContext context = this.delegate.newResourceContext(this.request, this.response);
-            context.authorize("create", "user." + name + ":domain", null);
+            String userDomainPrefix = System.getProperty(ZMSConsts.ZMS_PROP_USER_DOMAIN, ZMSConsts.USER_DOMAIN) + ".";
```
Comment (havetisyan) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSResources.java]: this is auto generated file so no changes are allowed here

**Comment by tatyano** [comment]
Henry,

Thank you for the review.
I'd like to talk about some changes but for now I'll just separate them and send another PR for ui first.

Thank you,
Chris

**Pull request closed without merging**